tests/benchdnn/common.cpp

```diff
@@ -243,12 +244,61 @@ static void zfree_protect(void *ptr) {
     ::free(ptr_start);
 }
 #endif
+struct memory_registry_t {
+    void add(void *ptr, size_t size) {
+        std::lock_guard<std::mutex> g(m_);
+        allocations_.emplace(std::pair<void *, size_t>(ptr, size));
```

Comment:
If by any chance `ptr` comes with the address already registered, it will be silently ignored. Maybe put an assert for `allocations_.find(ptr) == allocations_.end()`?